### ISSUE TITLE
Keras always predicts 0.0 probability for binary classification

### ISSUE THREAD

**Issue by extesy** [fchollet/keras#2746]
keras.py: https://gist.github.com/extesy/9a3f48fd760cb4d331386bd7542dba0b
training.tsv: https://raw.githubusercontent.com/jcarver989/kagglein-data/master/public/training.tsv?token=AAHLwMqsEeLm_lzai8-ZZ1aJVbsGasM9ks5XQ11kwA%3D%3D
test.tsv: https://raw.githubusercontent.com/jcarver989/kagglein-data/master/public/test.tsv?token=AAHLwC6BO3wgwFEVhiIeAXAyjrerzW7kks5XQ13WwA%3D%3D

Given the script above and two dataset files (training and testing datasets), the model is trained to about 70% accuracy, but model.predict_proba output is all zeroes.

I've tried it with both theano and tensorflow engines, but the result is still the same.

**Comment by duke7able** [comment]
what solution did you got due to which you closed the issue? can you please share I am also getting the same issue @extesy
